### PR TITLE
Update actuator Interface based on upstream changes

### PR DESCRIPTION
**Pull request by enxebre**
Bring cluster-api up to 0734939e05aeb64ab198e3feeee8b4e90ee5cbb2 https://github.com/kubernetes-sigs/cluster-api/commits/master
Which introduces changes to the actuator interface

Next changes are API breaking so we'll need to sync with the installer manifests https://github.com/kubernetes-sigs/cluster-api/commit/0cf3e178cb103ee1d2324918695fefe4bb09a1e0

### CODE REVIEW

**Comment by ingvagabund** [comment]
Should not extend the machine object (and potentially the cluster object) as well? https://github.com/kubernetes-sigs/cluster-api/pull/488/files#diff-c723d59aa09477303fd445f474e58de2R36

File: test/integration/create_update_delete_test.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 
 	kubernetesfake "k8s.io/client-go/kubernetes/fake"
 
+	"golang.org/x/net/context"
```
Comment (frobware) [inline test/integration/create_update_delete_test.go]: Why isn't this coming from just "context"?

Comment (ingvagabund) [reply test/integration/create_update_delete_test.go]: The PR: https://github.com/openshift/cluster-api-provider-aws/pull/118/file

**Pull request merged**